src/main/java/edu/harvard/iq/dataverse/engine/command/impl/ReconcileDatasetPidCommand.java

```diff
@@ -0,0 +1,181 @@
+@RequiredPermissions({})
+public class ReconcileDatasetPidCommand extends AbstractDatasetCommand<Dataset> {
+
+    private static final Logger logger = Logger.getLogger(ReconcileDatasetPidCommand.class.getName());
+    PidProvider newPidProvider;
+
+    public ReconcileDatasetPidCommand(DataverseRequest aRequest, Dataset theDataset, PidProvider newPidProvider) {
+        super(aRequest, theDataset);
+        this.newPidProvider = newPidProvider;
+    }
+
+
+    @Override
+    public Dataset execute(CommandContext ctxt) throws CommandException {
+        // ensure that only superuser can execute the command.
+        if (!(getUser() instanceof AuthenticatedUser) || !getUser().isSuperuser()) {
+            throw new PermissionException(BundleUtil.getStringFromBundle("admin.api.auth.mustBeSuperUser"), this,
+                    Collections.singleton(Permission.EditDataset), getDataset());
+        }
+        // Datast must be unreleased! This means there is only one version!
+        if (getDataset().isReleased()) { //@TODO: Clarify whether this is the best check...
+            throw new IllegalCommandException("Dataset already published, cannot alter PID Provider", this);
+        }
+        // Dataset must not be harvested!
+        if (getDataset().isHarvested()) {
+            throw new IllegalCommandException("Dataset is harvested, cannot alter PID Provider", this);
+        }
+        PidProvider currentPidProvider = getDataset().getEffectivePidGenerator();
+        // new PID Provider must be different to requested one!
+        if (this.newPidProvider.equals(currentPidProvider)) {
+            throw new IllegalCommandException("PID Provider " + currentPidProvider.getId() + " is same as configured. This Operation has no effect!", this);
+        }
+
+        GlobalId oldId = getDataset().getGlobalId();
+        if (oldId == null) {
+            throw new IllegalStateException("Dataset without a global identifier, cannot alter!");
+        }
+        logger.fine("Reconciling dataset( id =`" + getDataset().getId() + ")` - removing globalId `" + getDataset().getGlobalId() + '`');
+        // remove dataset PID
+        try {
+            if (currentPidProvider.alreadyRegistered(getDataset())) { //if not registered with PIDProvider than there is no need to delete it...
+                currentPidProvider.deleteIdentifier(getDataset()); // delete it externally
+            }
+            getDataset().setGlobalId(null); //  remove it internally
+            getDataset().setGlobalIdCreateTime(null);
+            getDataset().setIdentifierRegistered(false);
+        } catch (Exception e) {
+            logger.log(Level.WARNING, "Identifier deletion was not successful:", e.getMessage());
+        }
+
+        if(ctxt.systemConfig().isFilePIDsEnabledForCollection(getDataset().getOwner())) {
+            reconcileFilePids(ctxt, currentPidProvider);
+        }
+        getDataset().setPidGenerator(this.newPidProvider);
```

Comment:
This is setting the effective Pid Generator, which you probably don't want (or should do via the existing API call if you do).

Reply:
I believe I need to update this. Otherwise the dataset is an inconsistent state and new datafiles will not use the new PidGenerator.

Reply:
No you are right. This means the effective generator is obtained via the transitive closure